### PR TITLE
Doc updates

### LINKED ISSUE #27: Add documentation for Example Agents
we might want to have each example self-document and then aggregate that documentation programmatically.

### LINKED ISSUE #66: Add agent programming guide to docs

### PR DESCRIPTION
**Pull request by andyk**
* Expands Example Agents doc
* Adds an basic Agent Programming guide doc
* Updates docstrings in core.py
* Small updates to various other docs
* Adds a css file useful for custom styling
* Adds `--watch` and `--release` flags to `scripts\build_docs.py`
* Adds `sphinx-autobuild` to `dev-requirements`
* Updates documentation for developing agentos (mention `--release` flag)
* Fixes a copy+paste error where the github action that tests for up to date readme was called run_lint

Closes #27. Closes #66

### CODE REVIEW

**Comment by nickjalbert** [comment]
LGTM!

I think it'd be useful to update `documentation/includes/developing.rst` to describe how to use the new flags in `build_doc.py` for building and releasing the site.

**Comment by andyk** [comment]
@nickjalbert, are you planning on taking another review pass over this? Or should we merge it in?

File: documentation/programming_agents.rst [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+can think of as a simulation of how an agent might advance through the given
+environment, and what rewards it might receive along the way, if it were
+to use the given policy.
+
+Different types of agents and algorithms might use rollouts for
+different purposes, but rollouts always consist of the same basic structure.
+
+Since rollouts are used frequently and have a standard structure, AgentOS
+includes the ``agentos.core.rollout()`` utility function, but **note that the
+psuedocode above is a simplified version of ``agentos.core.rollout()``.**
```
Comment (nickjalbert) [inline documentation/programming_agents.rst:128]: typo: psuedocode -> pseudocode

**Review by nickjalbert — APPROVED** [review]
LGTM!  A couple of nits; feel free to fix and/or ignore as desired.

File: documentation/programming_agents.rst [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+============
+Environments can be either simulators (e.g., CartPole) or connectors to the
+real world (e.g., an environment for a chatbot that passes messages back and
+forth between the agent and other agents, humans, etc.).
+
+AgentOS does not define its own Environment API, instead we reuse `gym.Env`_.
+Environments must:
+
+* Descend from ``gym.Env``.
+* Define a function ``step(action) -> observation`` that takes an action and
```
[the remark this answers is not in the document]
Comment (andyk) [reply documentation/programming_agents.rst]: Done.

File: documentation/programming_agents.rst (hunk 1 above)
Comment (andyk) [reply documentation/programming_agents.rst:128]: Done.

**Pull request merged**